xbmc/addons/interfaces/platforms/android/General.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ *      Copyright (C) 2005-2018 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "addons/interfaces/General.h"
+#include "addons/kodi-addon-dev-kit/include/kodi/General.h"
+
+#include "platform/android/activity/XBMCApp.h"
+#include <sstream>
+
+using namespace kodi; // addon-dev-kit namespace
+
+namespace ADDON
+{
+
+char* Interface_General::get_platform_property(void* kodiBase, const char *name)
+{
+  char buffer[32];
+  if (strcmp(name,"jni_env") == 0)
+  {
+    //return / create JNI environment for the calling thread
+    sprintf(buffer, "%p", xbmc_jnienv());
+  }
+  else if (strcmp(name,"java_vm") == 0)
+  {
+    sprintf(buffer, "%p", CXBMCApp::get()->getActivity()->vm);
+  }
+  else if (strcmp(name,"sdk_version") == 0)
+  {
+    sprintf(buffer, "%d", CXBMCApp::get()->getActivity()->sdkVersion);
```

Comment:
Shouldn't we use `snprintf` to ensure no buffer overflow can happen?

Reply:
sure, that's an valid argument :-)

Reply:
no, wait, it's sprintf, there will never be more printed as type size in HEX

Reply:
If 32 char (current buffer size) is guaranteed to be large enough, I'm fine, but I suggest to write  a comment in the code explaining this.

Reply:
I'd still initialize buffer to zeros and use snprintf regardless.
If anyone is blindy adding new properties for example, additionally some compilers generate warnings if you use sprintf